src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,370 @@
+  <div>item 1</div>
+</div>
+```
+
+After adding second item (item 2):
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+  <div>item 2</div>
+</div>
+```
+![demo](/src/assets/htmx/savemultiple.gif)
+
+Adding multiple todo items is working now but on page load (e.g. if we refresh the page) the list of tasks are not being loaded.
+To make that work, we need to make a `GET` request.
+
+```html
+<div id="todo-list" hx-get="/taskList" hx-trigger="load"></div>
+```
+
+In the above HTML, I have added `hx-get` with value `/tasklist` and `hx-trigger` with value 'load'. The `load` value specifies when the request is sent. In this case it's when the HTML file is loaded.
+
+```go
+func listTask(w http.ResponseWriter, r *http.Request) {
+    tastList := getTODO() // gets data from database and returns HTML
+    io.WriteString(w, tastList)
+}
+func main() {
+    ...
+    http.HandleFunc("/save", listTask)
+    ...
+}
+```
+![demo](/src/assets/htmx/getalltask.gif)
+
+Now the list is shown even when the page is refreshed.
+
+For edit let's try something different instead of doing all our operations in a single we will be navigating to another page to do the edit.
+
+First let's make clicking on the todo item go to a different page that will contain a form for updating the todo item list.
+For that, we need to get an anchor tag instead of a div with the todo text inside a div.
+
+
+On the server side, I will get the task and return the proper form for updating that task.
+
+```go
+func main() {
+    ...
+    http.HandleFunc("/task/", individualTask)
+    ...
+}
+```
+
+Example response from server:
+
+```htmx
+  <div hx-boost="true">
+      <a href="/task/1">clean room</a>
+  </div>
+```
+
+Here the `hx-boost` attribute will replace the body element and add the previous page to the history entry.
+That means the back button on browsers will work perfectly.
+
+![demo](/src/assets/htmx/editpage.gif)
+
+For the actual editing, it will be similar to the other examples.
+
+```html
+<div>
+  <div>clean room</div>
+  <form id="myForm2" hx-put="/edit" hx-target="#app">
+    <input type="text" name="task" required="" />
+    <input type="hidden" name="old" value="1" />
+    <input type="submit" value="edit" />
+  </form>
+</div>
+```
+
+Now let's perform deletion on the list.
+
+First, we have to make a delete button.
+
+For that, I will update the list template and add a delete form:
+
+```html
+<div hx-boost="true" class="new-element">
+  <a href="/task/1">clean room</a>
+  <form
+    hx-trigger="click"
+    hx-delete="/delete"
+    hx-target="closest .new-element"
+    hx-swap="outerHTML swap:1s"
+  >
+    <input type="hidden" name="task" value="1" />
+    delete
+  </form>
+</div>
+```
+
+Most of the snippets in the delete form are similar to the other examples. But in the `hx-swap="outerHTML swap:1s"` attribute I added an extra value. This value will delay the swapping of the element by 1 second because I want to do a CSS animation during that time.
```

Comment:
```suggestion
Most of the snippets in the delete form are similar to the other examples. But in the `hx-swap="outerHTML swap:1s"` attribute, I added an extra value. This value will delay the swapping of the element by 1 second because I want to do a CSS animation during that time.
```